### PR TITLE
Move EnC debug information reading to Features.

### PR DESCRIPTION
**Pull request by tmat**
Encapsulate EnC debug info reading logic in EditAndContinueMethodDebugInfoReader and move it down to Features. Add tests.

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov

File: src/Compilers/Test/Utilities/CSharp/EditAndContinueTestUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        // Helpers that allow tests in IDE layers to inspect internals of EditAndContinueMethodDebugInformation without having IVT to the compiler.
+
+        public static int GetMethodOrdinal(this EditAndContinueMethodDebugInformation debugInfo)
+            => debugInfo.MethodOrdinal;
+
+        public static IEnumerable<string> InspectLocalSlots(this EditAndContinueMethodDebugInformation debugInfo)
+            => debugInfo.LocalSlots.IsDefault ? null : debugInfo.LocalSlots.Select(s => $"Offset={s.Id.SyntaxOffset} Ordinal={s.Id.Ordinal} Kind={s.SynthesizedKind}");
+
+        public static IEnumerable<string> InspectLambdas(this EditAndContinueMethodDebugInformation debugInfo)
+            => debugInfo.Lambdas.IsDefault ? null : debugInfo.Lambdas.Select(l => $"Offset={l.SyntaxOffset} Id={l.LambdaId.Generation}#{l.LambdaId.Ordinal} Closure={l.ClosureOrdinal}");
```
Comment (ivanbasov) [inline src/Compilers/Test/Utilities/CSharp/EditAndContinueTestUtilities.cs]: nit: the line is too long to be displayed on my screen without scrolling. maybe split it.

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                        customDebugInfo = info;
+                        foundAny = true;
+                    }
+                }
+
+                return foundAny;
+            }
+        }
+
+        public unsafe static EditAndContinueMethodDebugInfoReader Create(ISymUnmanagedReader5 symReader, int version)
```
Comment (ivanbasov) [inline src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs]: >public unsafe static EditAndContinueMethodDebugInfoReader Create(ISymUnmanagedReader5 symReader, int version) [](start = 8, length = 109)

nit: better move the public method above private classes

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    else
+                    {
+                        localSlots = lambdaMap = default;
+                    }
+
+                    return EditAndContinueMethodDebugInformation.Create(localSlots, lambdaMap);
+                }
+                catch (InvalidOperationException e) when (FatalError.ReportWithoutCrash(e)) // likely a bug in the compiler/debugger
+                {
+                    // TODO: CustomDebugInfoReader should throw InvalidDataException
```
Comment (ivanbasov) [inline src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs]: >TODO [](start = 23, length = 4)

Is this TODO for the current PR? or should we track it with a GitHub issue?

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs (hunk 2 above)
Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs]: We usually have classes first.

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs (hunk 3 above)
Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs]: This was there before. I don't think it's worth filing an issue at this point.

**Pull request merged**